src/applications/vaos/utils/selectors.js

```diff
@@ -366,8 +370,8 @@ export function getFacilityPageV2Info(state) {
     parentFacilitiesStatus,
     requestLocationStatus,
     selectedFacility: getChosenFacilityInfo(state),
-    singleValidVALocation: facilities?.length === 1,
-    showEligibilityModal: facilities?.length > 1 && showEligibilityModal,
+    singleValidVALocation: facilities?.length === 1 && !!data.vaFacility,
```

Comment:
If the user only has 1 valid facility, `data.vaFacility` should be set to it.  Check for this before setting `singleValidVALocation` to true, which should prevent `<VaFacilityInfoMessage>` from showing prematurely.